presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkBatchStreamReadersWithZstd.java

```diff
@@ -0,0 +1,256 @@
+    private static final DecimalType LONG_DECIMAL_TYPE = createDecimalType(30, 10);
+    private static final int ROWS = 10_000_000;
+    private static final int MAX_STRING = 10;
+    private static final List<?> NULL_VALUES = Collections.nCopies(ROWS, null);
+
+    @Benchmark
+    public Object readBlocksWithoutJni(BenchmarkData data)
+            throws Throwable
+    {
+        OrcBatchRecordReader recordReader = data.createRecordReader(false);
+        ImmutableList.Builder<Block> blocks = new ImmutableList.Builder<>();
+        while (recordReader.nextBatch() > 0) {
+            Block block = recordReader.readBlock(0);
+            blocks.add(block);
+        }
+        return blocks.build();
+    }
+
+    @Benchmark
+    public Object readBlocksWithJni(BenchmarkData data)
+            throws Throwable
+    {
+        OrcBatchRecordReader recordReader = data.createRecordReader(true);
+        ImmutableList.Builder<Block> blocks = new ImmutableList.Builder<>();
+        while (recordReader.nextBatch() > 0) {
+            Block block = recordReader.readBlock(0);
+            blocks.add(block);
+        }
+        return blocks.build();
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private final Random random = new Random(0);
+
+        private Type type;
+        private File temporaryDirectory;
+        private File orcFile;
+        private final OrcTester.Format format = DWRF;
+
+        @SuppressWarnings("unused")
+        @Param({
+                "boolean",
+                "tinyint",
+                "smallint",
+                "integer",
+                "bigint",
+                "decimal(10,5)",
+                "decimal(30,10)",
+                "timestamp",
+                "real",
+                "double",
+                "varchar_direct",
+                "varchar_dictionary",
+        })
+        private String typeSignature;
+
+        @SuppressWarnings("unused")
+        @Param({
+                "PARTIAL",
+                "NONE",
+                "ALL"
+        })
+        private Nulls withNulls;
+
+        @Setup
+        public void setup()
+                throws Exception
+        {
+            if (typeSignature.startsWith("varchar")) {
+                type = new TypeRegistry().getType(TypeSignature.parseTypeSignature("varchar"));
+            }
+            else {
+                type = new TypeRegistry().getType(TypeSignature.parseTypeSignature(typeSignature));
+            }
+
+            temporaryDirectory = createTempDir();
+            orcFile = new File(temporaryDirectory, randomUUID().toString());
+            writeOrcColumnPresto(orcFile, format, ZSTD, type, createValues());
+        }
+
+        @TearDown
+        public void tearDown()
+                throws IOException
+        {
+            deleteRecursively(temporaryDirectory.toPath(), ALLOW_INSECURE);
+        }
+
+        protected final List<?> createValues()
```

Comment:
nit: we don't usually use `final` on method. (Also, as a side note: we don't use `final` for tmp variables)